docs/scripting-ref.rst

```diff
@@ -2230,3 +2230,59 @@ Trigger mouse release event in web page.
 **Async:**: no.
 
 See notes about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-with-timeout:
+
+splash:with_timeout
+-------------------
+
+Run the function with the allowed timeout
+
+**Signature:** ``ok, result = splash:with_timeout(func, timeout)``
+
+**Parameters:**
+
+* func - the function to run
+* timeout - timeout, in seconds
+
+**Returns:** ``ok, result`` pair. If ``ok`` is not ``true`` then error happened during
+the function call or the timeout has passed; ``result`` provides an information
+about error type. If ``result`` is equal to ``timeout_over`` then specified timeout is elapsed.
+Otherwise, if ``ok`` is ``true`` then ``result`` contains the result of the executed function.
+
+**Async:** yes.
+
+Example 1:
+
+.. literalinclude:: ../splash/examples/with-timeout.lua
+   :language: lua
+
+
+Note that Splash can interrupt the running function only if some async function is called.
+Otherwise you should wait until the operation is finished.
```

Comment:
This paragraph might need to talk specifically about the fact that "timeout event can interrupt". And how one makes a function "async" without using splash methods by using `coro.yield`. It also might help to have asyncio-like comments in the code snippets saying "timeout event can fire here".